assets/javascripts/job_details.js

```diff
@@ -0,0 +1,207 @@
+		let alt = '';
+
+		if (step.name) {
+			alt = step.name;
+		}
+
+		if (step.is_parser_text_result) {
+			let elem = E('span', [], {'class': 'step_actions'});
+			elem.innerHTML = render_template(snippets.bug_actions,
+				tplargs);
+			elem = E('span', [elem, step.text_data],
+				{'class': 'resborder ' + step.resborder});
+			elem = E('span', [elem], {title: title,
+				'data-href': href, 'class': 'text-result',
+				onclick: 'toggleTextPreview(this)'});
+			stepnodes.push(E('div', [elem],
+				{'class': 'links_a text-result-container'}));
+			continue;
+		}
+
+		let url = render_template(snippets.module_url, tplargs);
+		let resborder = step.resborder;
+		let box = [];
+
+		if (step.screenshot) {
+			let thumb;
+
+			if (step.md5_dirname) {
+				thumb = render_template(snippets.md5thumb_url,
+					{DIRNAME: step.md5_dirname,
+					BASENAME: step.md5_basename});
+			} else {
+				thumb = render_template(snippets.thumbnail_url,
+					{FILENAME: step.screenshot});
+			}
+
+			if (step.properties &&
+				step.properties.indexOf('workaround') >= 0) {
+				resborder = 'resborder_softfailed';
+			}
+
+			box.push(E('img', [], {width: 60, height: 45,
+				'class': 'resborder ' + resborder,
+				src: thumb, alt: alt}));
+		} else if (step.audio) {
+			box.push(E('span', [], {alt: alt,
+				'class': 'icon_audio resborder ' + resborder}));
+		} else if (step.text && title == 'wait_serial') {
+			box.push(E('span', [], {alt: alt,
+				'class':'icon_terminal resborder '+resborder}));
+		} else if (step.text) {
+			box.push(E('span', [step.title ? step.title : 'Text'],
+				{'class': 'resborder ' + resborder}));
+		} else {
+			let content = step.title;
+
+			if (!content) {
+				content = E('i', [],
+					{'class': 'fas fa fa-question'});
+			}
+
+			box.push(E('span', [content],
+				{'class': 'resborder ' + resborder}));
+		}
+
+		stepnodes.push(E('div', [
+			E('div', [], {'class': 'fa fa-caret-up'}),
+			E('a', box, {'class': 'no_hover', title: title,
+				href: href, 'data-url': url})
+		], {'class': 'links_a'}));
+		stepnodes.push(' ');
+	}
+
+	let links = E('td', stepnodes, {'class': 'links'});
+
+	return E('tr', [component, result, links]);
+}
+
+function render_module_table(container, response) {
+	container.innerHTML = response.snippets.header;
+
+	if (response.modules === undefined || response.modules === null) {
+		return;
+	}
+
+	let E = create_element;
+	let thead = E('thead', [E('tr', [
+		E('th', ['Test']),
+		E('th', ['Result']),
+		E('th', ['References'], {style: 'width: 100%'})
+	])]);
+	let tbody = E('tbody');
+
+	container.appendChild(E('table', [thead, tbody],
+		{id: 'results', 'class': 'table table-striped'}));
+
+	for (let idx in response.modules) {
+		let module = response.modules[idx];
+
+		if (module.category) {
+			tbody.appendChild(E('tr', [E('td', [
+				E('i', [], {'class': 'fas fa-folder-open'}),
+				'\u00a0' + module.category
+			], {colspan: 3})]));
+		}
+
+		// FIXME: Add test module ID for per-row updates
```

Comment:
please don't add FIXME in code but rather keep that within a ticket

Reply:
This is a note to myself. Linking from a ticket to a specific point in code is kinda difficult.